### PR TITLE
EVG-19988: Update Go version, evergreen distros, test mongo version

### PR DESCRIPTION
**Pull request by johndaniels**

### CODE REVIEW

File: evergreen.yaml [hunk 1]
```diff
@@ -60,11 +60,19 @@ functions:
...
-        command: make get-mongodb
+          DECOMPRESS: ${decompress}
+    - command: subprocess.exec
+      type: setup
+      params:
+        command: make get-mongosh
+        working_dir: pail
+        include_expansions_in_env: ["MONGOSH_DOWNLOAD"]
+        env:
+          DECOMPRESS: ${decompress}
```
Comment (ybrill) [inline evergreen.yaml:75]: It seems like `DECOMPRESS` isn't used in this target.

File: evergreen.yaml [hunk 2]
```diff
@@ -138,56 +146,47 @@ tasks:
...
-
   - name: coverage
-    display_name: Coverage (Arch Linux)
+    display_name: Coverage
     expansions:
-      GOROOT: /opt/golang/go1.16
-      mongodb_url: http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-4.0.1.tgz
+      GOROOT: /opt/golang/go1.20
+      MONGODB_URL: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2204-6.0.6.tgz
+      MONGOSH_DOWNLOAD: curl https://downloads.mongodb.com/compass/mongosh-1.9.0-linux-x64.tgz -o mongosh.tgz && tar -xzvf mongosh.tgz
```
Comment (ybrill) [inline evergreen.yaml:154]: Would we want to use the other method (of having two `DECOMPRESS` variables) here too?

File: evergreen.yaml (hunk 1 above)
Comment (johndaniels) [reply evergreen.yaml]: Fixed

File: evergreen.yaml (hunk 2 above)
Comment (johndaniels) [reply evergreen.yaml]: Fixed

File: makefile [hunk 3]
```diff
@@ -143,16 +143,26 @@ $(buildDir)/output.%.lint: $(buildDir)/run-linter .FORCE
...
+	mkdir -p mongosh
+	cd mongosh && curl "$(MONGOSH_URL)" -o mongosh.tgz && $(MONGOSH_DECOMPRESS) mongosh.tgz && chmod +x ./mongosh-*/bin/*
+	cd mongosh && mv ./mongosh-*/bin/* .
 get-mongodb: mongodb/.get-mongodb
 	@touch $<
+get-mongosh: mongodb/.get-mongosh
+	@touch $<
 start-mongod: mongodb/.get-mongodb
-	./mongodb/mongod --dbpath ./mongodb/db_files
+	./mongodb/mongod --dbpath ./mongodb/db_files --port 27017 --replSet amboy
```
[the remark this answers is not in the document]
Comment (johndaniels) [reply makefile:158]: Ah good catch.

**Review by ybrill — APPROVED** [review]

**Pull request merged**